packages/@aws-cdk/aws-codebuild/lib/source.ts

```diff
@@ -19,6 +19,13 @@ export interface SourceConfig {
   readonly sourceProperty: CfnProject.SourceProperty;
 
   readonly buildTriggers?: CfnProject.ProjectTriggersProperty;
+
+  /**
+   * `AWS::CodeBuild::Project.SourceVersion`
+   * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codebuild-project.html#cfn-codebuild-project-sourceversion
+   * @default the latest version
+   */
+  readonly sourceVersion?: string | undefined;
```

Comment:
Just `sourceVersion?: string` is enough to be optional, no need for the `| undefined` 🙂.